meetings/2022-05-02.md

```diff
@@ -0,0 +1,118 @@
+* AC: In terms of recommendations, an application first needs to discover the capabilities of an OP: whether it supports DCR and client identifiers. If OP includes support for the `webid` scope, than it implements Solid-OIDC and supports client identifiers. In that case, a client (an app) should be using one of those client identifiers. If it doesn't, for example for testing, or wants to behave as ephermeral client, it can use DCR.
+* ...: For pod-browser it might be a bug.
+* eP: DCR is also problematic for client authorization. For instance, in the context of ACP, the recommended default is that there be no access. Unless the policy states "any client has access", a DCR-registered client would likely not be able to access anything.
+* ...: The Interop Panel places a heavy emphasis on using a stable identifier for clients: these identifiers are used to attach access controls
+* ...: in real-world scenarios, we should strongly recommend using client identifiers
+* AvdB: in the examples of Inrupt PodBrowser and Penny, they use DCR by default, even in the context of the same browser session. This overloads the issuer with useless clients. Using client id documents should be the general use case
+* AC: I agree. You may be noticing the interplay between implementations and specifications. Given that Solid-OIDC didn't have stable published version for so long, there have been multiple ways to implement authentication. Up until fairly recently the client id document wasn't boradly adopted. I believe Penny was implemented around a year ago. When it was originally written, client identifiers might not be well understood. Up until very recently, we haven't had a stable spec to point to. Now, I would be interested to see where the applications will go. Client IDs provide much stronger identity for the clients.
+* AvdB: Maybe penny isn't a best example but inrupt pod browser seems widely used by the community. Maybe post an issue on the Inrupt PodBrowser GitHub repository.
+* AC: Pod browser has very recently started to support client id documents. There are still cases where it doesn't work. For me the question is what can the AuthN panel do to improve the situation with implementations. We can have good patterns documenting how authentication flows work. This can include improving the primer. We could also provide better information about _why_ it is useful and explain why it is preffered over DCR. Nicolas is very involved with client tooling. One of the issues is that there are so many permutations of what servers support, and clients need to be able to navigate those permutations.
+* AvdB: Using the PB example since that was a common starting point. Currently, Client ID documents are a SHOULD not a MUST (same with DCR). There may be reasons not to force clients to use client id documents, how can we better encourage people to use these.
```

Comment:
```suggestion
* AvdB: Using the PB example since that was a common starting point. Currently, Client ID documents are a SHOULD not a MUST (same with DCR). There may be reasons not to force clients to use client id documents. How can we better encourage people to use these?
```